### PR TITLE
Update api access for enforced scopes

### PR DESCRIPTION
**Pull request by eriko**
While trying to set this up I noticed that the listed scopes that where needed were not accurate. Luckily the error log showed which were being requested.

As a sidenote should I start with the main 3.x version or should I issue the currently listed as released 2.2.8 version if I am just starting?

### CODE REVIEW

**Comment by bagofarms** [comment]
UDOIT 3.x is not "released" open source yet, but will be very soon.  We just have one pull request we're trying to finish first.  All work on UDOIT should be done on the 3.x code, which is in the `main` branch.  I already have a dev branch for UDOIT 3.1.0 (called `dev/v3-1-0`) that all feature PRs should point to.  I'll have someone review this and get it merged in.  Thanks!

File: INSTALL_CANVAS.md [hunk 1]
```diff
@@ -41,20 +41,49 @@ We strongly recommend you enforce scopes with your API key. The following scopes
...
 * Accounts
     * url:GET|/api/v1/accounts
-* Assignments
-    * url:GET|/api/v1/courses/:course_id/assignments
+    * url:GET|/api/v1/accounts/:id
 * Announcements
     * url:GET|/api/v1/announcements
+* Assignments
+    * url:GET|/api/v1/courses/:course_id/assignments
+    * url:get|/api/v1/courses/:course_id/assignments:id
```
Comment (webchuckweb) [inline INSTALL_CANVAS.md:49]:
```suggestion
    * url:GET|/api/v1/courses/:course_id/assignments/:id
```

File: INSTALL_CANVAS.md [hunk 2]
```diff
@@ -41,20 +41,49 @@ We strongly recommend you enforce scopes with your API key. The following scopes
...
+    * url:PUT|/api/v1/courses/:course_id/assignments:id
```
Comment (webchuckweb) [inline INSTALL_CANVAS.md:50]:
```suggestion
    * url:PUT|/api/v1/courses/:course_id/assignments/:id
```

File: INSTALL_CANVAS.md [hunk 3]
```diff
@@ -41,20 +41,49 @@ We strongly recommend you enforce scopes with your API key. The following scopes
...
 * Courses
     * url:GET|/api/v1/courses/:id
+    * url:OUT|/api/v1/courses/:id
```
Comment (webchuckweb) [inline INSTALL_CANVAS.md:53]:
```suggestion
    * url:PUT|/api/v1/courses/:id
```

File: INSTALL_CANVAS.md [hunk 4]
```diff
@@ -41,20 +41,49 @@ We strongly recommend you enforce scopes with your API key. The following scopes
...
+    * url:GET|/api/v1/courses/:course_id/modules_id/items
+    * url:GET|/api/v1/courses/:course_id/modules_id/items/:id
+    * url:PUT|/api/v1/courses/:course_id/modules/:module_id/items/:id
 * Pages
     * url:GET|/api/v1/courses/:course_id/pages
+    * url:GET|/api/v1/courses/:course_id/pages/:url
+    * url:GET|/api/v1/groups/:group_id/pages/:url
+    * url:PUT|/api/v1/courses/:course_id/pages/:url
+* Quiz Questions
+    * ul:GET|/api/v1/courses/:course_id/quizzes/:quiz_id/questions
```
Comment (webchuckweb) [inline INSTALL_CANVAS.md:77]:
```suggestion
    * url:GET|/api/v1/courses/:course_id/quizzes/:quiz_id/questions
```

File: INSTALL_CANVAS.md [hunk 5]
```diff
@@ -41,20 +41,49 @@ We strongly recommend you enforce scopes with your API key. The following scopes
...
 * Courses
     * url:GET|/api/v1/courses/:id
+    * url:OUT|/api/v1/courses/:id
+    * url:POST|/api/v1/courses_id/files
 * Discussion Topics
     * url:GET|/api/v1/courses/:course_id/discussion_topics
+    * url:GET|/api/v1/courses/:course_id/discussion_topics/:topic_id
+    * url:PUT|/api/v1/courses/:course_id/discussion_topics/:topic_id
+* Enrollment Terms
+    * url:GETurl:GET|/api/v1/accounts/:account_id/terms|/api/v1/accounts/:account_id/terms
```
Comment (webchuckweb) [inline INSTALL_CANVAS.md:60]:
```suggestion
    * url:GET|/api/v1/accounts/:account_id/terms
```

File: INSTALL_CANVAS.md [hunk 6]
```diff
@@ -41,20 +41,49 @@ We strongly recommend you enforce scopes with your API key. The following scopes
...
 * Files
     * url:GET|/api/v1/courses/:course_id/files
+    * url:GET|/api/v1/courses/:course_id/files/:id
 * Modules
     * url:GET|/api/v1/courses/:course_id/modules
+    * url:GET|/api/v1/courses/:course_id/modules/:id
+    * url:PUT|/api/v1/courses/:course_id/modules/:id
+    * url:GET|/api/v1/courses/:course_id/modules_id/items
```
Comment (webchuckweb) [inline INSTALL_CANVAS.md:68]:
```suggestion
    * url:GET|/api/v1/courses/:course_id/modules/:module_id/items
```

File: INSTALL_CANVAS.md [hunk 7]
```diff
@@ -41,20 +41,49 @@ We strongly recommend you enforce scopes with your API key. The following scopes
...
+    * url:GET|/api/v1/courses/:course_id/modules_id/items/:id
```
Comment (webchuckweb) [inline INSTALL_CANVAS.md:69]:
```suggestion
    * url:GET|/api/v1/courses/:course_id/modules/:module_id/items/:id
```

**Review by webchuckweb — CHANGES_REQUESTED** [review]
A few minor updates to the scope URLs.

**Review by AlanFCMV — CHANGES_REQUESTED** [review]
I tested this and you are still missing a scope on the list. Specifically:
- url:GET|/api/v1/accounts/:account_id/sub_accounts
Additionally, the following scope appears to not be necessary:
- url:POST|/api/v1/courses/:course_id/quizzes/:quiz_id/questions u

File: INSTALL_CANVAS.md [hunk 8]
```diff
@@ -41,20 +41,49 @@ We strongly recommend you enforce scopes with your API key. The following scopes
...
 * Announcements
     * url:GET|/api/v1/announcements
+* Assignments
+    * url:GET|/api/v1/courses/:course_id/assignments
+    * url:GET|/api/v1/courses/:course_id/assignments/:id
+    * url:PUT|/api/v1/courses/:course_id/assignments/:id
 * Courses
     * url:GET|/api/v1/courses/:id
+    * url:PUT|/api/v1/courses/:id
+    * url:POST|/api/v1/courses_id/files
```
[the remark this answers is not in the document]
Comment (eriko) [reply INSTALL_CANVAS.md:54]: "GET /authorize/check?error=invalid_scope&error_description=A requested scope is invalid, unknown, malformed, or exceeds the scope granted by the resource owner. The following scopes were requested, but not granted:
...
url:POST|/api/v1/courses/:course_id/files
...
Was in the error message I extracted from the logs. Was it removed in the last few days?

[the remark this answers is not in the document]
Comment (AlanFCMV) [reply INSTALL_CANVAS.md:54]: This is just a typo correction. It is currently listed as `url:POST|/api/v1/courses_id/files` when it should be `url:POST|/api/v1/courses/:course_id/files`

[the remark this answers is not in the document]
Comment (eriko) [reply INSTALL_CANVAS.md:54]: An another 'I can not typo' error.

**Comment by AlanFCMV** [comment]
Please add `url:GET|/api/v1/accounts/:account_id/sub_accounts` under accounts as well.

**Comment by eriko** [comment]
url:GET|/api/v1/accounts/:account_id/sub_accounts 
added

**Review by AlanFCMV — APPROVED** [review]

**Review by cidilabs — APPROVED** [review]

**Pull request merged**